REGRESSION (PHP to Python): http/tests/media/hls/hls-hdr-switch.html is a constant timeout

aproskuryakov opened the pull request:
#### ba207c023cfed1e0f9e94966d53ae782ba8b108e
<pre>
REGRESSION (PHP to Python): http/tests/media/hls/hls-hdr-switch.html is a constant timeout
<a href="https://bugs.webkit.org/show_bug.cgi?id=242559">https://bugs.webkit.org/show_bug.cgi?id=242559</a>
rdar://96780676

Reviewed by Jonathan Bedard.

* LayoutTests/http/tests/media/resources/hls/test-live.py: Fix the bug by only looking at the first value for each key,
not the whole list returned by parse_qs.
* LayoutTests/platform/mac/TestExpectations: Adjusted WebKit1 expectations, as this test passes on Big Sur, and
a different bug tracks problems on newer OSes.
* LayoutTests/platform/wk2/TestExpectations: Updated expectations to point to the right bug, and also separated
expectations for unrelated test hls-progress.html which is a flaky failure/crash.

Canonical link: <a href="https://commits.webkit.org/252342@main">https://commits.webkit.org/252342@main</a>
</pre>

JonWBedard commented on LayoutTests/platform/wk2/TestExpectations:
```diff
@@ -209,9 +209,9 @@ http/tests/download/inherited-encoding.html
-# webkit.org/b/221694
-http/tests/media/hls/hls-hdr-switch.html [ Timeout ]
-http/tests/media/hls/hls-progress.html [ Failure ]
+webkit.org/b/223280 http/tests/media/hls/hls-hdr-switch.html [ Timeout ]
```
How exactly did we find this issue, given this expectation?

aproskuryakov replied:
This test is subject to multiple issues, but as it wasn't skipped on wk1 yet, there was a different regression found. You can find more details in rdar://92303526.

JonWBedard reviewed (approved):
Approving this change because it's a clear progression, even if it doesn't fix all our problems.

aproskuryakov committed:
REGRESSION (PHP to Python): http/tests/media/hls/hls-hdr-switch.html is a constant timeout
https://bugs.webkit.org/show_bug.cgi?id=242559
rdar://96780676

Reviewed by Jonathan Bedard.

* LayoutTests/http/tests/media/resources/hls/test-live.py: Fix the bug by only looking at the first value for each key,
not the whole list returned by parse_qs.
* LayoutTests/platform/mac/TestExpectations: Adjusted WebKit1 expectations, as this test passes on Big Sur, and
a different bug tracks problems on newer OSes.
* LayoutTests/platform/wk2/TestExpectations: Updated expectations to point to the right bug, and also separated
expectations for unrelated test hls-progress.html which is a flaky failure/crash.

Canonical link: https://commits.webkit.org/252342@main
